tests/cupy_tests/linalg_tests/test_product.py

```diff
@@ -71,6 +71,40 @@ def test_dot_with_out(self, xp, dtype_a, dtype_b, dtype_c):
         return c
 
 
+@testing.parameterize(*testing.product({
+    'shape': [
+        ((2, 3, 4), (3, 4, 2)),
+        ((1, 1), (1, 1)),
+        ((1, 1), (1, 2)),
+        ((1, 2), (2, 1)),
+        ((2, 1), (1, 1)),
+        ((1, 2), (2, 3)),
+        ((2, 1), (1, 3)),
+        ((2, 3), (3, 1)),
+        ((2, 3), (3, 4)),
+        ((0, 3), (3, 4)),
+        ((2, 3), (3, 0)),
+        ((0, 3), (3, 0)),
+        ((3, 0), (0, 4)),
+        ((2, 3, 0), (3, 0, 2)),
+        ((0, 0), (0, 0)),
+        ((3,), (3,)),
+        ((2,), (2, 4)),
+        ((4, 2), (2,)),
+    ],
+}))
+@testing.gpu
+class TestCrossProduct(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_cross(self, xp, dtype_a, dtype_b):
+        shape_a, shape_b = self.shape_a
```

Comment:
This test does not run. 😛
Please confirm the test passes in your environment before receiving review.
More specifically you will need the following fixes:
* You need two dtype decorators  `@testing.for_all_dtypes(name='dtype_a')` `@testing.for_all_dtypes(name='dtype_b')`
* `self.shape_a` should be `self.shape`?